### PR TITLE
Fix request cycle

### PR DESCRIPTION
**Pull request by IonesioJunior**
# Pull Request Template

## Description
- [x] Remove pre-defined model id.
- [x] Save initial model checkpoint
- [x] Fix Cycle-Request event

## Type of change

Please mark options that are relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [X] I did follow the [contribution guidelines](https://github.com/OpenMined/PySyft/blob/master/CONTRIBUTING.md)
- [ ] New Unit tests added
- [ ] Unit tests pass locally with my changes

### CODE REVIEW

**Comment by cereallarceny** [comment]
Good to merge whenever you'd like @IonesioJunior .

@vvmnnnkv ?

File: gateway/app/main/processes/controller.py [hunk 1]
```diff
@@ -111,17 +112,17 @@ def delete_cycle(self, **kwargs):
...
                 worker_id: Worker's ID.
-                model_id: Model's ID.
+                name: Federated Learning Process Name.
                 version: Model's version.
             Return:
                 last_participation: Index of the last cycle assigned to this worker.
         """
-        _model = self._models.first(id=model_id)
-        _cycles = self._cycles.query(fl_process_id=_model.fl_process_id)
+        _fl_process = self._processes.first(name=name)
```
Comment (vvmnnnkv) [inline gateway/app/main/processes/controller.py]: Is version ignored? (That's fine for now, can be addressed later)

Comment (IonesioJunior) [reply gateway/app/main/processes/controller.py]: Done!

File: gateway/app/main/processes/controller.py [hunk 2]
```diff
@@ -136,19 +137,30 @@ def last_participation(self, worker_id: str, model_id: str, version: str) -> int
...
+                worker: Worker Object.
+                last_participation: The last time that this worker worked on this fl process.
+            Return:
+                last_participation: Index of the last cycle assigned to this worker.
+        """
         _accepted = False
 
-        # Retrieve model to track federated learning process
-        _model = self._models.first(id=model_id)
+        _fl_process = self._processes.first(name=name)
```
[the remark this answers is not in the document]
Comment (IonesioJunior) [reply gateway/app/main/processes/controller.py]: Done!

File: gateway/app/main/processes/controller.py [hunk 3]
```diff
@@ -237,13 +247,20 @@ def create_process(
...
+            raise FLProcessConflict
+
+        fl_process = self._processes.register(name=name, version=version)
+
+        # Register new model
+        _model = self._models.register(flprocess=fl_process)
 
-        # SAVE model weights into ModelCheckpoint
+        # Save model initial weights into ModelCheckpoint
+        # self._model_checkpoints.register(values=model, model=_model)
```
[the remark this answers is not in the document]
Comment (IonesioJunior) [reply gateway/app/main/processes/controller.py]: Done!

File: gateway/app/main/storage/models.py [hunk 4]
```diff
@@ -198,6 +198,7 @@ class FLProcess(db.Model):
     """ Federated Learning Process table.
         Columns:
             id (Integer, Primary Key): Federated Learning Process ID.
+            name (String) : Federated Process name.
```
[the remark this answers is not in the document]
Comment (IonesioJunior) [reply gateway/app/main/storage/models.py]: Done!

**Pull request merged**